### DISCUSSION TITLE
Livewire not loading the Model's attributes

### DISCUSSION THREAD

**Discussion by HeytalePazguato** [livewire/livewire#8757] [1. Help]
Hello everyone,

The post is a bit long because I have tried a lot of things but I hope someone could help me find out the issue.

I have a Livewire component, I upgraded from v2 to v3 and currently using v3.5.4; the issue started after upgrading.

My model `Vehicle` has three attributes, to keep it short I will just show one:

```
public function getVersionsQAttribute()
{
    $brands = Brand::where('brand', $this->brand)
        ->pluck('brand_ws')
        ->toArray();

    $sub_brands = SubBrand::where('sub_brand', $this->sub_brand)
        ->pluck('sub_brand_ws')
        ->toArray();

    $versions = QVehicle::where('year', '=', $this->year)
        ->whereIn('brand', $brands)
        ->whereIn('sub_brand', $sub_brands)
        ->pluck('version', 'amis')
        ->toArray();

    return $versions;
}
```

In the Livewire component, I have an instance of the model to edit

    public Vehicle $editing;

In the mount method, I have this:

```
public function mount()
{
    $this->editing = $this->makeBlank();
}
```

The `makeBlank()` method is this:

```
public function makeBlank()
{
    return new Vehicle();
}
```

In the Livewire component's view I am showing a table that has a button on each row to edit the vehicle, this opens a modal with some inputs and one of them is a `select` input:

```
<x-input.group for="createAmisQ-versions" label="VersionsQ" :error="$errors->first('editing.amis_q')" class="col-lg-2 col-3">
    <x-input.select wire:model="editing.amis_q" id="createAmisQ-versions" name="createAmisQ-versions">
        <option value=0 class="py-1">Select...</option>
        @foreach ($this->editing->versions_q as $amis => $version)
            <option wire:key="version-q-{{$amis}}" value="{{ $amis }}" class="py-1">{{ $version }}</option>
        @endforeach
    </x-input.select>
</x-input.group>
```

When I click on the edit button of any row, this is executed:

```
public function edit(Vehicle $vehicle)
{
    $this->useCachedRows();

    $this->editing = $vehicle;

    $this->showEditModal = true;
}
```

The table linked to the model already has data so `editing.amis_q` already has some data. When the modal is shown, the select input doesn't show any option selected, and all the `versions_q` are empty, this means only the "Select..." option is rendered.

If I close the modal and click in any row's edit, even in the same previously clicked button, then `versions_q` has all the versions, all the options are rendered, and the select input is shown with the selected option that contains `editing.amis_q`.

My problem is, that I do not understand why `editing.versions_q` is not rendering the options the first time the modal is opened but renders it correctly the second time the modal is opened.

I have tried to use `dd($this->editing->toArray())` inside of the `edit` method and every time I can see that `versions_q` exists and has the correct data, even on the first time the model is opened, so I am not sure why it is not rendered on the first time the modal is opened.

I have tried to create a computed Livewire property replacing the Model's attribute, like this

```
#[Computed()]
public function versions()
{
    $brands = Brand::where('brand', $this->editing->brand)
        ->pluck('brand_ws')
        ->toArray();

    $sub_brands = SubBrand::where('sub_brand', '=', $this->editing->sub_brand)
        ->orWhere('sub_brand', 'like', '%' . $this->editing->brand . '%')
        ->pluck('sub_brand_ws')
        ->toArray();

    $versions_q = Vehicle::where('year', '=', $this->editing->year)
        ->whereIn('brand', $brands)
        ->whereIn('sub_brand', $sub_brands)
        ->pluck('version', 'amis')
        ->toArray();

    return [
        'q' => $versions_q,
    ];
}
```

And modified the `@foreach` in the view, like this

    @foreach ($this->versions['q'] as $amis => $version)

But it still has the same issue, it only loads the second time the modal is opened

Lastly, I thought maybe the eloquent queries were taking too long, so I tried to optimize them to this:

```
#[Computed()]
public function versions()
{
    $versions_q = Vehicle::query()
        ->select('vehicles.version', 'vehicles.amis')
        ->join('brands', 'brands.brand_ws', '=', 'vehicles.brand')
        ->join('sub_brands', function($join) {
            $join->on('sub_brands.sub_brand_ws', '=', 'vehicles.sub_brand')
                ->where('sub_brands.sub_brand', '=', $this->editing->sub_brand)
                ->orWhere('sub_brands.sub_brand', 'like', '%'.$this->editing->brand.'%');
            })
        ->where('vehicles.year', '=', $this->editing->year)
        ->where('brands.brand', '=', $this->editing->brand)
        ->pluck('vehicles.version', 'vehicles.amis')
        ->toArray();

    return [
        'q' => $versions_q,
    ];
}
```

This provides the same output I had before but the problem is still present, the options are not loaded the first time the modal is opened and they load correctly the second time the modal is opened.

I have to add that after upgrading livewire I have configured the legacy model binding to keep the same functionality I had.

    'legacy_model_binding' => true,

If you have any clue why this is happening I would be grateful.

**Comment by lrljoe** [comment]
That's a lot of code!

It looks like you're setting editing to a new Model instance, then setting it to an existing Model later down the line.

Remember that a Computed Property is generated once per lifecycle.

If you create the instance with the existing Model, it should work fine.  Use some logic and optional parameters to either 
Create a new instance
or
Load an existing instance.

To be honest, when it comes to Modals, just use the Wire Elements Modals package, it works a dream.

**Reply by HeytalePazguato** [reply]
Yes, sorry for the long post but I was trying to show and explain everything I have done so far. Originally it was working with the Model attributes but now it only works after the second time the modal is opened. The strange thing is, that I have 25+ pages with the same composition, a table with rows, and each row can be edited through a modal.

This is the only one that doesn't load the elements on the first time but it loads them on the second time the modal is opened.

I will explore the Wire Elements Modal package, but I still would like to know why it loads only after the second time the modal is opened.
